### PR TITLE
Interledger: Explain Like I'm Five (ELI5)

### PR DESCRIPTION
**Pull request by professorhantzen**
Following discussions with @emschwartz I've drafted this (hopefully accurate) explanation of ILP concepts in reddit's "ELI5" tradition.  The goal of an ELI5 is to aim to present concepts simply enough to be understood by an interested five year old, such that an adult can use the document to gain a practical understanding of a complex system in a short amount of time.

### CODE REVIEW

**Comment by professorhantzen** [comment]
Apologies for the github n00bness - this PR is not intended to modify anything already in the RFC's. It's meant to just add one new file.

**Comment by adrianhopebailie** [comment]
@professorhantzen it looks like if you rebase you'll get rid of the unwanted changes.

File: 0000-interledger-eli5/0000-interledger-eli5.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+3) Connectors can each charge a little bit for their part in handling the payment.  After the chain is built, the connector closest the sender explains how much the receiver will get if the payment goes down that chain.  If it's much less than what the sender wants to send, the sender will probably choose a different connector.  So connectors try make the cost to the sender as little as possible, while still earning something.
+
+4) The sender decides whether or not to use that connector, based on how much it will cost.  If it's a "no", the sender can start over asking a different connector, or asking the same one again (who may figure out a better way this time).  When the sender is happy with the cost, the payment can go ahead.
+
+For each payment, a chain with links of trust is formed between the sender, connector - or connectors - and the receiver.  Each link in the chain trusts the one before it not to waste its time or run away with the assets, but no link needs to trust any other links in the chain or even know about them.  What's great about this, is it means the sender and the receiver don't need to trust each other, or even know who each other is.
```
Comment (emschwartz) [inline 0000-interledger-eli5/0000-interledger-eli5.md]: Technically it's not that you trust the previous link in the chain not to run away with the assets (assuming links = connectors), you trust the ledger to faithfully keep the other person's funds on hold

File: 0000-interledger-eli5/0000-interledger-eli5.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+1) A sender asks a connector: "Do you know this receiver?"
+
+2) If the connector knows the receiver, great.  If not, the connector talks to its "peers" (its "connector friends") to see if any know the receiver, or know another connector that might know the receiver... and so on.  Through this process, a chain of connectors linking the sender to the receiver is discovered.  This chain may involve just one connector, or it may involve many.
+
+3) Connectors can each charge a little bit for their part in handling the payment.  After the chain is built, the connector closest the sender explains how much the receiver will get if the payment goes down that chain.  If it's much less than what the sender wants to send, the sender will probably choose a different connector.  So connectors try make the cost to the sender as little as possible, while still earning something.
+
+4) The sender decides whether or not to use that connector, based on how much it will cost.  If it's a "no", the sender can start over asking a different connector, or asking the same one again (who may figure out a better way this time).  When the sender is happy with the cost, the payment can go ahead.
```
Comment (emschwartz) [inline 0000-interledger-eli5/0000-interledger-eli5.md]: All of these steps technically describe the quoting phase, not the "Preparation" phase. Normally when we talk about payments being prepared, that means the money is being put on hold all the way down the path

File: 0000-interledger-eli5/0000-interledger-eli5.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+**Payment Security**
+
+Even with all the preparation, there are still a couple of things that could go wrong.  For example, what happens if a connector lies? There are a couple of ways connectors could choose to either be annoying and cause delays, or even try and steal assets.  It's unlikely, but if it did happen, ILP guarantees the sender will get their original asset back in their account.  Then of course, the sender could try sending again a different way.  In a small number of cases, this situation may result in a connector next to the bad one losing money.  That's why connectors charge a little bit for making payments, and it's also why they want to choose carefully to link up with only good connectors.  Also, if a connector did lie, all connectors will know about it and "unfriend" them to make it difficult for that connector to be involved in an ILP payment again.
```
Comment (emschwartz) [inline 0000-interledger-eli5/0000-interledger-eli5.md]: In the case where you have proper ledgers in between each pair of connectors, connectors can only get screwed by the ledger, not by other connectors

Comment (emschwartz) [inline 0000-interledger-eli5/0000-interledger-eli5.md]: It's unclear what connectors lying means in this context. Does that mean they promised a rate they couldn't deliver upon?

File: 0000-interledger-eli5/0000-interledger-eli5.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+Much of what happens throughout an ILP payment is actually only there to protect the asset from exactly this kind of thing during its transition from one ledger to another.  One of the main things used is called a _cryptographic condition_.  This is a complex topic, so we're going to use an analogy to represent it.  Instead of a "cryptographic condition", let's call this thing a _special box._
+
+*The Special Box*
```
Comment (emschwartz) [inline 0000-interledger-eli5/0000-interledger-eli5.md]: 👍

File: 0000-interledger-eli5/0000-interledger-eli5.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+- "I'm going to give someone on your ledger ¥1050 JPY.  Can you take ¥1050 from my account, and put it in your special 'holding' account?
+- "Then, can you take these boxes, keep one, and give the other to the receiver?"
+- "Finally, wait for a password from the receiver - if the password opens your box, put the ¥1050 in the receivers account, tell the receiver you've done that, and let me know too!"
+
+3) The ledger manager does all this, and gives the receiver the last box.  The receiver hears all about the payment and puts the password they previously arranged with the sender into the box.  The box opens!  So, the receiver knows this box came from the sender, and that this payment is a real payment the sender wants to make.  The receiver also knows that right now, some Japanese Yen are on hold for the receiver in the JPY ledger, waiting for this password.  The receiver wants the money, and trusts the JPY ledger manager, so gives the JPY ledger manager the password.
```
Comment (emschwartz) [inline 0000-interledger-eli5/0000-interledger-eli5.md]: In the beginning of this paragraph, specify this is the JPY ledger manager

File: 0000-interledger-eli5/0000-interledger-eli5.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+4) The JPY ledger manager puts the password in the box, and it opens! The ledger manager is now sure that the box came from the original sender, and that the receiver and the sender agreed and are happy with whats happening. The JPY ledger manager changes the entries in the ledger, so that the holding account doesn't have the ¥1050 JPY, and the receiver's account has it instead.  The JPY ledger manager tells the connector what happened, and because the JPY ledger manager trusts the connector, gives the connector the password.
```
Comment (emschwartz) [inline 0000-interledger-eli5/0000-interledger-eli5.md]: The ledger manager actually doesn't care that the box came from the sender, just that their customer (the connector) will be okay with the transfer going ahead

Comment (emschwartz) [inline 0000-interledger-eli5/0000-interledger-eli5.md]: Also the ledger manager doesn't "trust" the connector, it's just that the connector has an account on that ledger and because money was transferred from them they're entitled to know why

File: 0000-interledger-eli5/0000-interledger-eli5.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+5) The connector tries the password in the box, and of course - it opens.  The connector now knows this password will open the box the USD ledger manager has, and the ledger manager will put $10 USD into the connector's account on the USD ledger.  The connector trusts the USD ledger manager, and so gives the password to them.
```
Comment (emschwartz) [inline 0000-interledger-eli5/0000-interledger-eli5.md]: Might be helpful to re-emphasize here that the password is the same for all the boxes

File: 0000-interledger-eli5/0000-interledger-eli5.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+
+_Note: The reason for the "holding" part is that it protects the assets.  If anything goes wrong with the payment while the assets are on hold - for example, if the receiver rejects the payment - the assets can simply be taken from the holding accounts and put back into the accounts they came from.  In fact, in the case of the example given, if the receiver rejected the payment the assets would **have** to be put back.  Because if the receiver rejected the payment, no password would have been given that could open the boxes, so none of the ledger managers would have done anything._
+
+
+**Asset Transformation & Connectors**
```
Comment (emschwartz) [inline 0000-interledger-eli5/0000-interledger-eli5.md]: I wouldn't call this "transformation" but "exchange". The assets stay the same, but are exchanged along the way for other assets

File: 0000-interledger-eli5/0000-interledger-eli5.md [hunk 9]
```diff
@@ -0,0 +1,124 @@
...
+
+The Internet can be seen as a set of rules about how to send messages. For example, when a person follows the rules of the Internet to send an email, it can be said that the person is _using the Internet Protocol to transmit information._
+
+An _asset_ is a word meaning anything valuable. For example, money is an asset.
+
+Interledger (aka _Interledger Protocol_, or _ILP_) is a set of rules about how to send assets. For example, when a person follows the rules of the Interledger to send money, it can be said that the person is _using the Interledger Protocol to transmit an asset._
+
+Interledger, like the Internet, isn't only used by people. It can be used by a computer, a group of computers, or a group of people and/or computers acting together as a company or corporation. The word _party_ is sometimes used to describe a person, thing, or any group of people or things that can _participate in an action._
+
+The Interledger Protocol has four basic building blocks that participate together in the action of transmitting an asset from one party to another.
```
Comment (emschwartz) [inline 0000-interledger-eli5/0000-interledger-eli5.md]: It might be good to include in the intro a little more of the benefits of what ILP gives you (payments between people on different networks who use different currencies)

**Comment by emschwartz** [comment]
Replaced by https://github.com/interledger/rfcs/pull/212
